src/sql/workbench/contrib/query/browser/gridPanel.ts

```diff
@@ -736,7 +742,14 @@ export abstract class GridTableBase<T> extends Disposable implements IView {
 	private rebuildActionBar() {
 		let actions = this.getCurrentActions();
 		this.actionBar.clear();
-		this.actionBar.push(actions, { icon: true, label: false });
+		if (this.options.displayActions) {
+			this.actionBar.push(actions, { icon: true, label: false });
+		}
+	}
+
+	public showActionBar(display: boolean) {
```

Comment:
I think I left the same comment in the other PR you have.

1. let's use getter/setter, showActionBar(boolean) seems weird to me.
2. the name should be consistent, lets use showActionBar in the options too.
3. only rebuild the action bar when the value is actually changed.

```TypeScript
public get showActionBar():boolean {
    return this.options.showActionBar;
}

public set showActionBar(v: boolean) {
    if(this.options.showActionBar !== v) {
        this.options.showActionBar = v;
        this.rebuildActionBar();
    }
}
```